Fixed Issue #30

FunNoober opened the pull request:
To fix issue #30, of custom controls not working on Fedora 34/KDE, I simply created a new scene that is ran when ever the project is launched. When it does this, it gets the operating system and if you are running an X11 or Server system, then it disables the custom controls, and enables the window manager default controls.

FunNoober committed:
Fixed Issue #30

To fix issue #30, of custom controls not working on Fedora 34/KDE, I simply created a new scene that is ran when ever the project is launched. When it does this, it gets the operating system and if you are running an X11 or Server system, then it disables the custom controls, and enables the window manager default controls.

martin-piliar commented:
Hey! We are glad that you took the time to work on this bug fix. @lentsius-bark nor I have a way to test it at the moment, so I'll summon @Pomichal, who will hopefully take a look at it sometime when he has a pinch of free time.

lentsius-bark commented:
@Pomichal this is awesome, give it a test, also thanks to @martin-piliar for bringing this up, it can finally move on forward

lentsius-bark commented on Scripts/TopArea.gd:
```diff
@@ -32,7 +37,7 @@ func _on_mouse_exited() -> void:
 
 func _input(event: InputEvent) -> void:
-	if event is InputEventMouseButton:
+	if event is InputEventMouseButton and ProjectSettings.get_setting("display/window/size/borderless") == true:
```
rather than doing it this way, would it be possible to completely turn off the _input function? `set_process_input(false)` that way we wouldn't need to process this at all, since the WM of the OS would take care of it.

FunNoober replied:
I did not exactly think of doing it but after testing on my machine, this seems to work!

lentsius-bark commented on StartDialog.gd:
```diff
@@ -0,0 +1,9 @@
+		get_tree().change_scene("res://Main.tscn")
+	else:
+		ProjectSettings.set_setting("display/window/size/borderless", true)
+		get_tree().change_scene("res://Main.tscn")
```
1. just a matter of style here, I'd move the changing of scene outside of the if/else block as it gets triggered regardless in the same way. 
2. since I can't test this I don't know how well this works, however I often made use of : [this](https://docs.godotengine.org/en/stable/classes/class_os.html#class-os-property-window-borderless) instead, as the project setting can stay the same, what matters is the window.

*As mentioned however this might work the same, if it does, then all good.*

FunNoober replied:
Using `OS.set_borderless_window()` is a lot cleaner and on my machine, it appears to work the same way.

FunNoober committed:
Cleaner Code

- 'Using OS.get_borderless_window()' for if statements resulting in cleaner code
- No longer checking if the window is borderless in the '_input' function, instead disabling the '_input' function.
- In the start dialog, the changing of the scene is called outside the if/else block.

lentsius-bark commented:
@FunNoober you made these changes super quickly! thanks! Now we just need @Pomichal to give it a quick test and it's ready for a merge

Pomichal commented:
@FunNoober thanks for working on this! It seems a little bit like a "hack", but I can't think of any better solution neither, so it's time to merge it!
